### DISCUSSION TITLE
Preview service, customize and exclude previews from backup

### DISCUSSION THREAD

**Discussion by lab-at-nohl** [tine20/tine20#7482] [General]
At least since Tine 2022.12.1 the **Preview**-service came more in the focus. Users are prompted the preview before download choice. 

While you can discuss about mail attachements I really like it for documents though. I found metaway's docker image, It does the job, however, it is very large (you can remove some hundred MBs easyly without breaking soffice). In that regard I have some questions: 

- Under what licence does the PHP script for serving the preview file come? 
I would like to have a smaller docker image where the admin can "inject" LibreOffice as AppImage (the rest of the docker image stays stable but you can choose any AppImage version you like). Also but with more effort, it would be nice to have soffice loaded once and [convert documents by UNO interface](https://github.com/unoconv/unoserver). Can I copy your PHP code for my purpose? 
- Do you know how to add hyphenation to converted preview documents? 
- Also I expect a **lot more files in Tine**. Can I separate preview files from originals in the VFS? 
When doing backups I'd like to exclude the preview images, in case they are lost I wold recreate them. 
- Maybe, is there an option to only have "live" previous instead of create images?
- What does the pdf-Option I saw in the code do? The DMS in my office creates previews as pdf files, which is very helpful and modern browsers can open them as fast as graphics... Is it this?

Best
Johannes

**Comment by pschuele** [comment]
hi Johannes, sorry for the late reply.

I forwarded your questions to the team and hopefully someone will answer them soon.
at the moment, there is no way to separate the previews from the original files in the VFS, but that sounds like a legit feature request.

CU
Philipp

**Comment by pschuele** [comment]
Hi Johannes,
here are the answers to your questions - some of them might be worth discussing in detail in a video call (I'll send you an email).

> Under what licence does the PHP script for serving the preview file come?

AGPL V3

> I would like to have a smaller docker image where the admin can "inject" LibreOffice as AppImage

sounds like a good idea - we would need to discuss this further and make a plan.

> Do you know how to add hyphenation to converted preview documents?

sorry, no idea :(

> Can I separate preview files from originals in the VFS?

we would suggest to create a separate tree for the previews for separation. that should also be considered by the tine --backup command.

> "live" previews instead of create images

that's not possible at the moment.

> What does the pdf-Option I saw in the code do?

it's used to create certain PDF-exports. we could also think about, creating pdfs for all document-previews instead of images. we should also discuss, which preview-service we'll support in the long run (alternatives: OnlyOffice, https://gotenberg.dev) we could also make this configurable and use the tine preview-service as a backup.

**Comment by pschuele** [comment]
btw: we just published the code in github -> https://github.com/tine-groupware/document-preview-service
